### PR TITLE
deleteElementTree must handle DefinitionContainer

### PR DESCRIPTION
**Pull request by swwilson-bsi**
ElementTreeWalker and deleteElementTree did not recognize that a nested DefinitionContainer holds definition elements

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: core/backend/src/ElementTreeWalker.ts [hunk 1]
```diff
@@ -50,6 +50,8 @@ enum ElementPruningClassification { PRUNING_CLASS_Normal = 0, PRUNING_CLASS_Subj
 
 function classifyElementForPruning(iModel: IModelDb, elementId: Id64String): ElementPruningClassification {
   const el = iModel.elements.getElement(elementId);
+  if (el instanceof DefinitionContainer)
+    return ElementPruningClassification.PRUNING_CLASS_DefinitionPartition;
   return (el instanceof Subject) ? ElementPruningClassification.PRUNING_CLASS_Subject :
     (el instanceof DefinitionElement) ? ElementPruningClassification.PRUNING_CLASS_Definition :
       (el instanceof DefinitionPartition) ? ElementPruningClassification.PRUNING_CLASS_DefinitionPartition :
```
Comment (nick4598) [inline core/backend/src/ElementTreeWalker.ts:57]: Could you group these as below?

```suggestion
      (el instanceof DefinitionPartition || el instanceof DefinitionContainer) ? ElementPruningClassification.PRUNING_CLASS_DefinitionPartition :
```

Comment (nick4598) [reply core/backend/src/ElementTreeWalker.ts:57]: Feel free to resolve both I left if not. Just wanted to make it possible for the suggestions to be committed from github and have it be good to go

Comment (swwilson-bsi) [reply core/backend/src/ElementTreeWalker.ts:57]: No. DefinitionContainer is-a DefinitionElement, so the (el instanceof DefinitionElement) ? ElementPruningClassification.PRUNING_CLASS_Definition :  case would grab it.

Comment (nick4598) [reply core/backend/src/ElementTreeWalker.ts:57]: Ah got it. Could you add a comment explaining that?

File: core/backend/src/ElementTreeWalker.ts [hunk 2]
```diff
@@ -50,6 +50,8 @@ enum ElementPruningClassification { PRUNING_CLASS_Normal = 0, PRUNING_CLASS_Subj
 
 function classifyElementForPruning(iModel: IModelDb, elementId: Id64String): ElementPruningClassification {
   const el = iModel.elements.getElement(elementId);
+  if (el instanceof DefinitionContainer)
+    return ElementPruningClassification.PRUNING_CLASS_DefinitionPartition;
```
[the remark this answers is not in the document]
Comment (swwilson-bsi) [reply core/backend/src/ElementTreeWalker.ts:54]: See above

File: core/backend/src/ElementTreeWalker.ts (hunk 1 above)
Comment (swwilson-bsi) [reply core/backend/src/ElementTreeWalker.ts:60]: done

Comment (nick4598) [reply core/backend/src/ElementTreeWalker.ts:60]: thank you

**Review by nick4598 — APPROVED** [review]

**Pull request merged**